src/DynamoPackagesUI/Utilities/PackageManagerCommands.cs

```diff
@@ -0,0 +1,485 @@
+    /// </summary>
+    internal class PackageManagerCommands : CefCommands
+    {
+        public PackageManagerCommands(DynamoViewModel dynamoViewModel, PackageLoader model, PackageManagerViewModel pkgManagerViewModel) : 
+            base(dynamoViewModel, model, pkgManagerViewModel)
+        {
+        }
+
+        public List<string> PackagesToInstall { get; set; }
+
+
+        private dynamic _downloadRequest;
+        public dynamic DownloadRequest
+        {
+            get { return _downloadRequest; }
+            set
+            {
+                if (value is Newtonsoft.Json.Linq.JObject)
+                    _downloadRequest = value;
+                else
+                    _downloadRequest = JsonConvert.DeserializeObject<dynamic>(value);
+            }
+        }
+
+        private dynamic _pkgRequest;
+        public dynamic PkgRequest
+        {
+            get { return _pkgRequest; }
+            set
+            {
+                if (value is Newtonsoft.Json.Linq.JObject)
+                    _pkgRequest = value;
+                else
+                    _pkgRequest = JsonConvert.DeserializeObject<dynamic>(value);
+            }
+        }
+
+        private string PackageInstallPath { get; set; }
+
+        public void GoToWebsite()
+        {
+            dynamoViewModel.PackageManagerClientViewModel.GoToWebsite();
+        }
+
+        /// <summary>
+        /// Install Dynamo Package
+        /// </summary>
+        /// <param name="downloadPath"></param>
+        private void InstallPackage(string downloadPath)
+        {
+            var firstOrDefault = Model.LocalPackages.FirstOrDefault(pkg => pkg.ID == DownloadRequest.asset_id.ToString());
+            if (firstOrDefault != null)
+            {
+                var dynModel = dynamoViewModel.Model;
+                try
+                {
+                    firstOrDefault.UninstallCore(dynModel.CustomNodeManager, dynamoViewModel.Model.GetPackageManagerExtension().PackageLoader, dynModel.PreferenceSettings);
+                }
+                catch
+                {
+                    MessageBox.Show(String.Format(Resources.MessageFailToUninstallPackage,
+                        dynamoViewModel.BrandingResourceProvider.ProductName,
+                        DownloadRequest.asset_name.ToString()),
+                        Resources.UninstallFailureMessageBoxTitle,
+                        MessageBoxButton.OK, MessageBoxImage.Error);
+                }
+            }
+
+            var settings = dynamoViewModel.Model.PreferenceSettings;
+            PackageDownloadHandle packageDownloadHandle = new PackageDownloadHandle();
+            packageDownloadHandle.Name = DownloadRequest.asset_name;
+            packageDownloadHandle.Done(downloadPath);
+
+            string installedPkgPath = string.Empty;
+            if (packageDownloadHandle.Extract(dynamoViewModel.Model, this.PackageInstallPath, out installedPkgPath))
```

Comment:
Why don't you use the other Extract method which already provides you Package object.